akka-actor-typed/src/main/scala/akka/actor/typed/internal/Supervision.scala

```diff
@@ -48,7 +53,9 @@ import akka.util.unused
 @InternalApi
 private abstract class AbstractSupervisor[O, I, Thr <: Throwable](strategy: SupervisorStrategy)(
     implicit ev: ClassTag[Thr])
-    extends BehaviorInterceptor[O, I] {
+    extends BehaviorInterceptor[Any, I] {
+
+  // FIXME test that supervision works also when exception from a non-O message type, like InternalProtocol in EventSourcedBehavior
```

Comment:
Add such test or ticketify

Reply:
added test in a1cd6d4fbd96a95db5ee6332a4e607ea6282375d

supervision interceptors are of type Any so intercepting everything